### ISSUE TITLE
if tf.reduce_sum(y_hat, 1) == self.token2idx["<pad>"]: break

### ISSUE THREAD

**Issue by gaoxuezhao** [Kyubyong/transformer#144]
I don't understand the conditional sentence in the loop when infer, it is always false。

if tf.reduce_sum(y_hat, 1) == self.token2idx[""]: break

**Comment by ZhichaoOuyang** [comment]
Yes, I have this question too.

**Comment by tracyfly** [comment]
+1

**Comment by BetterZH** [comment]
+1

**Comment by GuoshenLi** [comment]
you are right. the author is wrong. I also think the following "decoder_inputs = tf.concat((decoder_inputs, y_hat), 1)" is wrong. cuz what we need to concat is not the total y_hat but the last time step of y_hat. so y_hat[:,-1]
